### PR TITLE
benchmark: Add benchmark report for TiDB 4.0 on DBaaS, TPC-H

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What is changed, added or deleted? (Required)

Add benchmark report for TiDB v4.0 on DBaaS.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by JaySon-Huang** [comment]
@shuke987 @scsldb PTAL

**Comment by ran-huang** [comment]
@JaySon-Huang Please add the doc in `TOC.md`

File: benchmark/tpch-dbaas.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    --host ${tidb_host} --port 4000 --db tpch_10 --password ${password} \
+    --sf 10 \
+    --tiflash \
+    --analyze --tidb_build_stats_concurrency 8 --tidb_distsql_scan_concurrency 30
+```
+
+运行完之后确认数据已经建立好 TiFlash 副本
+
+```
+mysql> select * from information_schema.tiflash_replica;
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:64]: 请提供一个用户可以直接复制的命令

File: benchmark/tpch-dbaas.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+| tpch_10      | region     |       56 |             1 |                 |         1 |        1 |
+| tpch_10      | orders     |       72 |             1 |                 |         1 |        1 |
+| tpch_10      | customer   |       68 |             1 |                 |         1 |        1 |
+| tpch_10      | lineitem   |       75 |             1 |                 |         1 |        1 |
++--------------+------------+----------+---------------+-----------------+-----------+----------+
+```
+
+### 执行查询
+
+本文会比较智能选择、TiKV隔离读、TiFlash隔离读三种[读取模式](/reference/tiflash/use-tiflash.md#使用-tidb-读取-tiflash)下的性能。使用 mysql-client 连接到 TiDB 集群，先设置优化参数以及读取的模式。
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:81]:
```suggestion
本文会比较智能选择、TiKV 隔离读、TiFlash 隔离读等三种[读取模式](/reference/tiflash/use-tiflash.md#使用-tidb-读取-tiflash)下的 TiDB 性能。使用 MySQL 客户端连接到 TiDB 集群，先设置优化参数以及读取的模式。
```

File: benchmark/tpch-dbaas.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+```
+mysql> set @@session.tidb_allow_batch_cop = 1; set @@session.tidb_opt_distinct_agg_push_down = 1; set @@tidb_distsql_scan_concurrency = 30;
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:84]: 请提供一个用户可以直接复制的命令

File: benchmark/tpch-dbaas.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+# 可以通过以下语句确认当前变量值
+mysql> select @@tidb_isolation_read_engines , @@tidb_allow_batch_cop , @@tidb_opt_distinct_agg_push_down , @@tidb_distsql_scan_concurrency;
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+| @@tidb_isolation_read_engines | @@tidb_allow_batch_cop | @@tidb_opt_distinct_agg_push_down | @@tidb_distsql_scan_concurrency |
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+| tikv,tiflash                  | 1                      | 1                                 | 30                              |
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+```
+
+然后执行 TPC-H 的查询语句。你可以在这里找到具体的查询语句：[tidb-bench/tpch/queries](https://github.com/pingcap/tidb-bench/tree/master/tpch/queries)
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:97]:
```suggestion
然后执行 TPC-H 的查询语句。你可以在这里找到具体的查询语句：[tidb-bench/tpch/queries](https://github.com/pingcap/tidb-bench/tree/master/tpch/queries)。
```

File: benchmark/tpch-dbaas.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+### 测试结果
+
+| Query ID |  智能选择  |  TiKV隔离读  |  TiFlash隔离读  |
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:101]:
```suggestion
| Query ID |  智能选择  |  TiKV 隔离读  |  TiFlash 隔离读  | 
```

File: benchmark/tpch-dbaas.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+| 15      |      1.08s |       8.02s |         1.00s |
+| 16      |      1.95s |       2.34s |         1.27s |
+| 17      |      8.45s |      15.20s |         8.79s |
+| 18      |      ?.??s |       ?.??s |         ?.??s |
+| 19      |      3.24s |       5.61s |         3.29s |
+| 20      |      1.39s |       4.20s |         9.51s |
+| 21      |      6.63s |       7.74s |        14.61s |
+| 22      |      2.50s |       3.12s |         2.63s |
+
+![TPC-H Query Result](/media/tpch-query-result-v4.0-dbaas.png)
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:126]: 图片最好提供英文的

File: benchmark/tpch-dbaas.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+说明：
+
+- 图中蓝色为智能选择，红色为 TiKV 隔离读，黄色为 TiFlash 隔离读，纵坐标是 Query 的处理时间，越低越好
+- 图中不显示 Query 18 结果，因 4.0.0-rc.1 版本 TiDB 聚合算子在大数据量情况下消耗较多内存，Query 18 查询时会报 "Lost connection to MySQL server during query"。相关 Issue [tidb#14103](https://github.com/pingcap/tidb/issues/14103)、[tidb#14413](https://github.com/pingcap/tidb/issues/14413)
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:131]:
```suggestion
- 图中不显示 Query 18 结果，因为 4.0.0-rc.1 版本 TiDB 聚合算子在大数据量情况下消耗较多内存，Query 18 查询时会报错：`Lost connection to MySQL server during query`。相关 Issue 见 [tidb#14103](https://github.com/pingcap/tidb/issues/14103)、[tidb#14413](https://github.com/pingcap/tidb/issues/14413)。
```

File: benchmark/tpch-dbaas.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+| 19      |      3.24s |       5.61s |         3.29s |
+| 20      |      1.39s |       4.20s |         9.51s |
+| 21      |      6.63s |       7.74s |        14.61s |
+| 22      |      2.50s |       3.12s |         2.63s |
+
+![TPC-H Query Result](/media/tpch-query-result-v4.0-dbaas.png)
+
+说明：
+
+- 图中蓝色为智能选择，红色为 TiKV 隔离读，黄色为 TiFlash 隔离读，纵坐标是 Query 的处理时间，越低越好
```
Comment (ran-huang) [inline benchmark/tpch-dbaas.md:130]:
```suggestion
- 图中蓝色为智能选择，红色为 TiKV 隔离读，黄色为 TiFlash 隔离读，纵坐标是 Query 的处理时间，纵坐标越低，性能越好。
```

**Comment by JaySon-Huang** [comment]
@ran-huang addressed, PTAL again

File: benchmark/tpch-dbaas-v4.0.md [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+-- 智能选择:'tikv,tiflash', TiKV隔离读:'tikv', TiFlash隔离读:'tiflash'.
+set @@session.tidb_isolation_read_engines = 'tikv,tiflash';
+-- 可以通过以下语句确认当前变量值
+select @@tidb_isolation_read_engines , @@tidb_allow_batch_cop , @@tidb_opt_distinct_agg_push_down , @@tidb_distsql_scan_concurrency;
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+| @@tidb_isolation_read_engines | @@tidb_allow_batch_cop | @@tidb_opt_distinct_agg_push_down | @@tidb_distsql_scan_concurrency |
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+| tikv,tiflash                  | 1                      | 1                                 | 30                              |
++-------------------------------+------------------------+-----------------------------------+---------------------------------+
+```
```
Comment (ran-huang) [inline benchmark/tpch-dbaas-v4.0.md]: How about this?
- Add a `copyable` code snippet
- Separate the command and the result
---

{{< copyable "sql" >}}

```sql
set @@session.tidb_allow_batch_cop = 1;
set @@tidb_opt_agg_push_down = 1;
set @@session.tidb_opt_distinct_agg_push_down = 1;
set @@tidb_distsql_scan_concurrency = 30;
-- 设置读取模式. 不同读取模式对应的值如下:
-- 智能选择:'tikv,tiflash', TiKV隔离读:'tikv', TiFlash隔离读:'tiflash'.
set @@session.tidb_isolation_read_engines = 'tikv,tiflash';
-- 可以通过以下语句确认当前变量值
select @@tidb_isolation_read_engines , @@tidb_allow_batch_cop , @@tidb_opt_distinct_agg_push_down , @@tidb_distsql_scan_concurrency;
```

```
+-------------------------------+------------------------+-----------------------------------+---------------------------------+
| @@tidb_isolation_read_engines | @@tidb_allow_batch_cop | @@tidb_opt_distinct_agg_push_down | @@tidb_distsql_scan_concurrency |
+-------------------------------+------------------------+-----------------------------------+---------------------------------+
| tikv,tiflash                  | 1                      | 1                                 | 30                              |
+-------------------------------+------------------------+-----------------------------------+---------------------------------+
```

File: TOC.md [hunk 10]
```diff
@@ -14,6 +14,7 @@
     - [TPC-C 性能对比 - v4.0 对比 v3.0](/benchmark/v4.0-performance-benchmarking-with-tpcc.md)
     - [TPC-C 性能对比 - v3.0 对比 v2.1](/benchmark/v3.0-performance-benchmarking-with-tpcc.md)
     - [线上负载与 `Add Index` 相互影响测试](/benchmark/online-workloads-and-add-index-operations.md)
+    - [TiDB DBaaS TPC-H 10G 性能测试报告](/v4.0-benchmark-on-dbaas.md)
```
Comment (yikeke) [inline TOC.md:17]:
```suggestion
    - [TiDB DBaaS TPC-H 10G 性能测试报告](/benchmark/v4.0-benchmark-on-dbaas.md)
```
To fix a link

**Pull request closed without merging**